[ESIMD] Fix minor bug in ballot SFINAE filtering.

kbobrovs opened the pull request:
It lead to the following warning below and allowing unintended types:
...sycl/ext/intel/esimd/math.hpp:685:22: warning: address of function 'detail::is_type<unsigned short, cl::sycl::ext::intel::esimd::ushort, cl::sycl::ext::intel::esimd::uint>' will always
      evaluate to 'true' [-Wpointer-bool-conversion]
    std::enable_if_t<detail::is_type<T, ushort, uint> && (N > 0 && N <= 32),
                     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ ~~

Signed-off-by: Konstantin S Bobrovsky <konstantin.s.bobrovsky@intel.com>

sndmitriev reviewed (approved):
Thanks for fixing it!

kbobrovs commented:
All test failures are unrelated

- Unexpectedly Passed Tests (1):
  SYCL :: KernelAndProgram/undefined-symbol.cpp
addressed in https://github.com/intel/llvm-test-suite/pull/824

- Failed Tests (2):
  SYCL :: api/simd_any_all.cpp
  SYCL :: regression/big_const_initializer.cpp
addressed in https://github.com/intel/llvm-test-suite/pull/958

- Failed Tests (1) (Linux):
      SYCL :: Assert/assert_in_simultaneously_multiple_tus.cpp
address in https://github.com/intel/llvm-test-suite/pull/959
- Failed Tests (2) (Windows):
      SYCL :: Regression/pi_release.cpp
      SYCL :: Scheduler/HostAccDestruction.cpp
These two are infra errors (HostAccDestruction error is the same):
```
$ "FileCheck" "W:\jenkins-dir\workspace\SYCL_CI\intel\Win\Test_Suite\llvm-test-suite\SYCL\Regression\pi_release.cpp"
# command stderr:
Could not create process (q:\lref1\ref\lit\tools\windows\filecheck.exe) due to [WinError 999] Error performing inpage operation
error: command failed with exit status: 127
```
